kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,257 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
     labeled_dataset = torch.utils.data.Subset(dataset, tindices)
     unlabeled_dataset = torch.utils.data.Subset(dataset, uindices)
     return labeled_dataset, unlabeled_dataset
+
+
+def domain_balanced_split(domain_labels, n_partitions, split_ratios):
+    """Get indices of random split. Samples with the same domain label will be split based on the given ratios.
```

Comment:
Is this kind of stratified sampling too?: https://en.wikipedia.org/wiki/Stratified_sampling
Is `domain-balanced` your creation or existing terminology? If self-creation, consider whether using stratified will be clearer to others.

Reply:
Yes, stratified is better. Thank you.